p2p/agent.go

```diff
@@ -193,6 +207,7 @@ func NewAgent(cfg Config, chainID uint32, genesisHash hash.Hash256, broadcastHan
 		unicastInboundAsyncHandler: unicastHandler,
 		reconnectTimeout:           cfg.ReconnectInterval,
 		qosMetrics:                 NewQoS(time.Now(), 2*cfg.ReconnectInterval),
+		batchManager:               newBatchManager(),
```

Comment:
try to a few parameters to batchManager:
id: agent.host.HostIdentity(),
broadcastMsg: agent.broadcastMsg,
unicastMsg: agent.UnicastMsg,
opts...

Then you may move the `batchFactory` functionality into batchManager.